src/test/ui/lint/unused/unused-result.stderr

```diff
@@ -18,7 +18,7 @@ LL |     foo::<MustUseMsg>();
    |
    = note: some message
 
-error: unused result
+error: unused result of type `isize`
```

Comment:
Alternatively, the type could be displayed in a `note`, but using the title is consistent with `unused_must_use` and easier to scan IMO.